### ISSUE TITLE
BUG: read from cache instead of raising for invalid arguments passed to metrics

### ISSUE THREAD

**Issue by cakedev0** [probabl-ai/skore#2491]
### Describe the bug

For `CrossValidationReport`, `report.metrics.accuracy(X=X, y=y)` doesn't raise if `report.metrics.accuracy()` was computed (and cached before), and it returns this cached value.

### Steps/Code to Reproduce

```Python
from sklearn.datasets import load_iris
from sklearn.ensemble import GradientBoostingClassifier
from skore import CrossValidationReport
from report_artifacts import log_cv_report_to_mlflow

X, y = load_iris(return_X_y=True, as_frame=True)

report = CrossValidationReport(GradientBoostingClassifier(random_state=0), X, y)
try:
    report.metrics.accuracy(X=X, y=y)
except ValueError as e:
    print(e)  # prints an error
report.metrics.accuracy()
report.metrics.accuracy(X=X, y=y)  # should raise but doesn't
```

### Expected Behavior

The second `report.metrics.accuracy(X=X, y=y)` should raise.

### Actual Behavior

It doesn't raise and returns result for the test dataset ignoring X & y.

### Environment

```shell
System:
    python: 3.12.11 (main, Aug 18 2025, 19:19:11) [Clang 20.1.4 ]
executable: /home/arthur/probabl/skore/skore/.venv/bin/python
   machine: Linux-6.17.0-14-generic-x86_64-with-glibc2.39

Python dependencies:
        skore: 0.0.0+unknown
          pip: 26.0.1
       jinja2: 3.1.6
       joblib: 1.5.3
   matplotlib: 3.10.8
        numpy: 2.4.2
       pandas: 2.3.3
       plotly: 6.5.2
         rich: 14.3.2
 scikit-learn: 1.8.0
      seaborn: 0.13.2
 skore[local]: None
        skrub: 0.7.2
skore-hub-project: None
```
